src/Symfony/Component/String/BinaryString.php

```diff
@@ -0,0 +1,440 @@
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\String;
+
+use Symfony\Component\String\Exception\ExceptionInterface;
+use Symfony\Component\String\Exception\InvalidArgumentException;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ * @author Hugo Hamon <hugohamon@neuf.fr>
+ *
+ * @throws ExceptionInterface
+ */
+class BinaryString extends AbstractString
+{
+    public function __construct(string $string = '')
+    {
+        $this->string = $string;
+    }
+
+    public static function fromRandom(int $length = 16): self
+    {
+        $string = '';
+
+        do {
+            $string .= str_replace(['/', '+', '='], '', base64_encode(random_bytes($length)));
+        } while (\strlen($string) < $length);
+
+        return new static(substr($string, 0, $length));
+    }
+
+    public function byteCode(int $index = 0): ?int
+    {
+        $str = $offset ? $this->slice($index, 1) : $this;
+
+        return '' === $str->string ? null : \ord($str->string);
+    }
+
+    public function append(string ...$suffix): parent
+    {
+        $str = clone $this;
+        $str->string .= 1 >= \count($suffix) ? $suffix[0] ?? '' : implode('', $suffix);
+
+        return $str;
+    }
+
+    public function chunk(int $length = 1): array
+    {
+        if (1 > $length) {
+            throw new InvalidArgumentException('The maximum length of each segment must be greater than zero.');
+        }
+
+        if ('' === $this->string) {
+            return [];
+        }
+
+        $chunks = [];
+
+        foreach (str_split($this->string, $length) as $chunk) {
+            $chunks[] = $str = clone $this;
+            $str->string = $chunk;
+        }
+
+        return $chunks;
+    }
+
+    public function endsWith($suffix): bool
+    {
+        if ($suffix instanceof parent) {
+            $suffix = $suffix->string;
+        } elseif (!\is_string($suffix)) {
+            return parent::endsWith($suffix);
+        }
+
+        return \strlen($this->string) - \strlen($suffix) === ($this->ignoreCase ? strripos($this->string, $suffix) : strrpos($this->string, $suffix));
+    }
+
+    public function equalsTo($string): bool
+    {
+        if ($string instanceof parent) {
+            $string = $string->string;
+        } elseif (!\is_string($string)) {
+            return parent::equalsTo($string);
+        }
+
+        if ($this->ignoreCase) {
+            return 0 === strcasecmp($string, $this->string);
+        }
+
+        return $string === $this->string;
+    }
+
+    public function folded(): parent
+    {
+        $str = clone $this;
+        $str->string = strtolower($str->string);
+
+        return $str;
+    }
+
+    public function indexOf($needle, int $offset = 0): ?int
+    {
+        if ($needle instanceof parent) {
+            $needle = $needle->string;
+        } elseif (!\is_string($needle)) {
+            return parent::indexOf($needle);
```

Comment:
```suggestion
            return parent::indexOf($needle, $offset);
```

(same in `indexOfLast` and maybe in other sub classes?)